### PR TITLE
Firebase ML Node.js SDK Structure

### PR DESCRIPTION
**Pull request by ifielker**
Firebase ML Node.js SDK Structure. (Implementation to follow)

### CODE REVIEW

File: src/index.d.ts [hunk 1]
```diff
@@ -5170,6 +5203,138 @@ declare namespace admin.messaging {
...
+      readonly models: Model[];
+      readonly pageToken?: string;
+    }
+
+    /**
+     * A TFLite Model output object
+     */
+    interface TFLiteModel {
+      readonly sizeBytes: number;
+
```
Comment (hiranya911) [inline src/index.d.ts]: Remove redundant space. Here and below.

File: src/index.d.ts [hunk 2]
```diff
@@ -5170,6 +5203,138 @@ declare namespace admin.messaging {
...
+    }
+
+    /**
+     * A Firebase ML Model input object
+     */
+    interface ModelOptions {
+      displayName?: string;
+      tags?: string[];
+
+      tfLiteModel?: {gcsTFLiteUri?: string;};
```
Comment (hiranya911) [inline src/index.d.ts]: I think `gcsTFLiteUri` should be required.

File: src/machine-learning/machine-learning.ts [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+import {FirebaseError} from '../utils/error';
+
+import * as validator from '../utils/validator';
+
+// const ML_HOST = 'mlkit.googleapis.com';
+
+/**
+ * Internals of an ML instance.
+ */
+class MlInternals implements FirebaseServiceInternalsInterface {
```
Comment (hiranya911) [inline src/machine-learning/machine-learning.ts]: MachineLearningInternals

File: src/machine-learning/machine-learning.ts [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+  }
+}
+
+/**
+ * A TFLite Model output object
+ */
+export interface TFLiteModel {
+  readonly sizeBytes: number;
+
+  readonly gcsTfliteUri?: string;
```
Comment (hiranya911) [inline src/machine-learning/machine-learning.ts]: Required?

File: src/machine-learning/machine-learning.ts [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+
+
+/**
+ * A Firebase ML Model input object
+ */
+export class ModelOptions {
+  public displayName?: string;
+  public tags?: string[];
+
+  public tfLiteModel?: { gcsTFLiteUri?: string; };
```
Comment (hiranya911) [inline src/machine-learning/machine-learning.ts]: Shouldn't `gcsTFLiteUri` be required?

File: src/index.d.ts (hunk 2 above)
Comment (ifielker) [reply src/index.d.ts]: Sure. We can require it for now. It is the only option in a oneof. Eventually there will be more options. Then they will all be optional. (But at least one will be required.)

File: src/machine-learning/machine-learning.ts (hunk 3 above)
Comment (ifielker) [reply src/machine-learning/machine-learning.ts]: done

File: src/machine-learning/machine-learning.ts (hunk 4 above)
Comment (ifielker) [reply src/machine-learning/machine-learning.ts]: Sure.

File: src/machine-learning/machine-learning.ts (hunk 5 above)
Comment (ifielker) [reply src/machine-learning/machine-learning.ts]: Sure.

File: src/index.d.ts (hunk 1 above)
Comment (ifielker) [reply src/index.d.ts]: done

Comment (hiranya911) [reply src/index.d.ts]: I was actually referring to the redundant newlines between entries. Sorry for not being explicit.

**Pull request merged**